In [ ]:
# DATA ANALYSIS

In [11]:
import pandas as pd

In [12]:
data = pd.read_csv('WorldHappiness.csv').dropna()
print(data)
print()
print(f"Data shape: {data.shape}")
print()
print(f"Data shape: {data.dtypes}")

     Country name  year  Life Ladder  Log GDP per capita  Social support  \
0     Afghanistan  2008        3.724               7.350           0.451   
1     Afghanistan  2009        4.402               7.509           0.552   
2     Afghanistan  2010        4.758               7.614           0.539   
3     Afghanistan  2011        3.832               7.581           0.521   
4     Afghanistan  2012        3.783               7.661           0.521   
...           ...   ...          ...                 ...             ...   
2358     Zimbabwe  2019        2.694               7.698           0.759   
2359     Zimbabwe  2020        3.160               7.596           0.717   
2360     Zimbabwe  2021        3.155               7.657           0.685   
2361     Zimbabwe  2022        3.296               7.670           0.666   
2362     Zimbabwe  2023        3.572               7.679           0.694   

      Healthy life expectancy at birth  Freedom to make life choices  \
0              

Which countries or regions are the happiest in a particular factor ?

- Economic factors:
    Log GDP per capita
- Social factors:
    Social support
    Generosity
    Perceptions of corruption
- Health Factors:
    Healthy life expectancy at birth
- Psychological and Emotional Factors:
    Life Ladder
    Positive affect
    Negative affect
- Autonomy and Freedom:
    Freedom to make life choices

We will be test our data based on the average values, so here are the generally accepted mean for required factors:

Life Ladder: Around 5 to 6
Log GDP per capita: Around 9 to 10
Social Support: Around 0.8 to 0.9
Healthy Life Expectancy at Birth: Around 60 to 70 years
Freedom to Make Life Choices: Around 0.7 to 0.8
Generosity: Around 0.2 to 0.3
Perceptions of Corruption: Around 0.3 to 0.5
Positive Affect: Around 0.6 to 0.7
Negative Affect: Around 0.2 to 0.3

In [25]:
for column in data.columns[2:]:
    column_mean = data.groupby("Country name")[column].mean()

    country_with_highest_column_mean = column_mean.idxmax()
    highest_column_mean = column_mean.max()

    print(
        f"The country with the highest mean of {column} is {country_with_highest_column_mean} with a mean of {highest_column_mean:.3f}.")

The country with the highest mean of Life Ladder is Denmark with a mean of 7.643.
The country with the highest mean of Log GDP per capita is Luxembourg with a mean of 11.643.
The country with the highest mean of Social support is Iceland with a mean of 0.978.
The country with the highest mean of Healthy life expectancy at birth is Japan with a mean of 73.634.
The country with the highest mean of Freedom to make life choices is Norway with a mean of 0.951.
The country with the highest mean of Generosity is Myanmar with a mean of 0.603.
The country with the highest mean of Perceptions of corruption is Romania with a mean of 0.945.
The country with the highest mean of Positive affect is Panama with a mean of 0.822.
The country with the highest mean of Negative affect is Iraq with a mean of 0.504.


Now we can compare and understand why these exact countries are the best performers in one or another happiness factor:

Denmark has even higher average score of 7.6 for a Life Ladder than worldwidly accepted.
Luxembourg robust its economy growth significantly and exceed the basic score by 1.6.
Iceland have very good social support which is higher than in other countries.
Japan is known as a high-tech development country as well as high-quality healthcare provider, that`s why average life expectancy is really high and is about 73 years.
Norway's exceptionally high mean for "Freedom to make life choices" of 0.951 can be attributed to a sense of autonomy and empowerment among its citizens.
Myanmar's high average generosity is a reflection of the country's significant religious and cultural emphasis on altruism and communal support, which is frequently connected to Buddhist traditions.
Romania's high average of views of corruption suggests that the public is generally aware of and concerned about corruption in public institutions and the government, which has an effect on governance and confidence.
 Panama's thriving culture, social connectivity, and generally upbeat mood among its people are the main reasons for the country's high mean positive affect.
Iraq's high mean negative affect is a reflection of the country's population's recent decades-long stress, aggression, and instability.

What is the happiest country in each group of factors ?

In [14]:
# ECONOMIC FACTORS

country_mean = data.groupby("Country name")[
    "Log GDP per capita"].mean()

happiest_country = country_mean.idxmax()
happiest_country_total = country_mean.max()

print(
    f"The happiest country based on economic factors is {happiest_country} with an average score of {happiest_country_total:.3f}.")

The happiest country based on economic factors is Luxembourg with an average score of 11.643.


In [24]:
# SOCIAL FACTORS

social_factors = ["Social support", "Generosity", "Perceptions of corruption"]

country_mean = data.groupby("Country name")[social_factors].mean()

country_mean["mean"] = country_mean.mean(axis=1)

happiest_country = country_mean["mean"].idxmax()
happiest_country_mean = country_mean["mean"].max()

print(
    f"The happiest country based on social factors is {happiest_country} with an average of {happiest_country_mean:.3f}.")

The happiest country based on social factors is Thailand with an average of 0.720.


Strong social support networks that offer a dependable network of family and community, along with a generous culture where actions of giving and assisting others are deeply ingrained, are responsible for Thailand's high levels of happiness. In addition, despite corruption problems, the Thai people's resilience, capacity for joy in everyday life, and participation in group events and festivities all add to their general contentment.

In [ ]:
# Optionally, print the detailed sums for each country
print("\nDetailed average for each country:")
print(country_mean)

In [16]:
# HEALTH FACTORS

country_mean = data.groupby("Country name")[
    "Healthy life expectancy at birth"].mean()

happiest_country = country_mean.idxmax()
happiest_country_total = country_mean.max()

print(
    f"The happiest country based on health factors is {happiest_country} with an average of {happiest_country_total:.3f}.")

The happiest country based on health factors is Japan with an average of 73.634.


In [17]:
# Psychological and Emotional Factors

social_factors = ["Life Ladder", "Positive affect", "Negative affect"]

country_mean = data.groupby("Country name")[social_factors].mean()

country_mean["Total"] = country_mean.mean(axis=1)

happiest_country = country_mean["Total"].idxmax()
happiest_country_total = country_mean["Total"].max()

print(
    f"The happiest country based on psychological and emotional factors is {happiest_country} with an average of {happiest_country_total:.3f}.")

The happiest country based on psychological and emotional factors is Denmark with an average of 2.876.


Due to its strong "Life Ladder" score—which is a reflection of its good quality of life, strong social services, and stable economy—Denmark has the highest position in terms of happiness. The fact that Danes rarely feel negative emotions and frequently experience good ones adds to their general well-being. Additionally, high levels of "Positive affect" and low levels of "Negative affect" show this.

In [ ]:
# Optionally, print the detailed sums for each country
print("\nDetailed average for each country:")
print(country_mean)

In [18]:
# Autonomy and Freedom

country_sums = data.groupby("Country name")[
    "Freedom to make life choices"].mean()

happiest_country = country_sums.idxmax()
happiest_country_total = country_sums.max()

print(
    f"The happiest country based on autonomy and freedom factors is {happiest_country} with an average score of {happiest_country_total:.3f}.")

The happiest country based on autonomy and freedom factors is Norway with an average score of 0.951.


What is the happiest country ?

In [26]:
country_mean = data.groupby("Country name")[data.columns[2:]].mean()

country_mean['Average'] = country_mean.mean(axis=1)

happiest_country = country_mean['Average'].idxmax()
highest_total_sum = country_mean['Average'].max()

happiest_country_mean = country_mean.loc[happiest_country, data.columns[2:]]

print(
    f"The happiest country is {happiest_country} with an average of {highest_total_sum:.3f}.")
print()

for column, value in happiest_country_mean.items():
    print(f"{column}: {value:.3f}")

The happiest country is Iceland with an average of 10.446.

Life Ladder: 7.461
Log GDP per capita: 10.877
Social support: 0.978
Healthy life expectancy at birth: 71.852
Freedom to make life choices: 0.927
Generosity: 0.254
Perceptions of corruption: 0.696
Positive affect: 0.807
Negative affect: 0.165


Potential Causes of Happiness in Iceland

Social support: Tight communities that place a high value on friendships and family make for strong social support networks.

High Standard of Living: Access to first-rate healthcare, education, and other necessities is guaranteed by economic stability and a high GDP per capita.

Good Governance: Effective public services and justice are ensured by low levels of corruption and high levels of trust in governmental institutions.

Health & Well-Being: Long, healthy lives are the result of high life expectancy and quality healthcare services.

Freedom and Autonomy: Encouraging individual autonomy and self-determination enables people to make meaningful life decisions.

Natural Environment: Access to nature and outdoor activities improve well-being. Stunning scenery and a clean environment contribute to this.

Cultural factors: Happiness as a whole is influenced by cultures that value innovation, resiliency, and balanced lifestyles.
